tests/server-integration/Common/BootServer.hs

```diff
@@ -0,0 +1,101 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io>
+--
+-- SPDX-License-Identifier: MPL-2.0
+
+module Common.BootServer
+  ( unit_run_with_cmd_option_port
+  , unit_run_with_env_var_port
+  , unit_run_with_no_port_specified
+  , unit_run_with_cmd_opt_and_bad_env_var_port
+  , unit_run_with_bad_cmd_opt_and_env_var_port
+  , unit_run_with_bad_env_var_port
+  , unit_run_with_bad_cmd_option_port
+  , unit_run_with_bad_str_cmd_option_port) where
+
+import Control.Concurrent.Async (async, cancel, poll)
+import Control.Exception (SomeException(SomeException))
+import System.Environment (setEnv, unsetEnv, withArgs)
+import System.Time.Extra (sleep)
+import Test.Tasty.HUnit (assertFailure, (@=?))
+import Web.Main (RunServerException(..), runServer)
+
+testPort :: String
+testPort = "8079"
+
+testServer :: String -> [String] -> IO () -> IO () -> (String -> IO ()) -> IO ()
+testServer envPort args serverRunning serverStopped serverCrashed = do
+  setEnv "COFFER_SERVER_PORT" envPort
+  server <- async $ withArgs args runServer
+  sleep 1
+  serverStatus <- poll server
+  case serverStatus of
+    Nothing -> cancel server >> serverRunning
+    Just (Right _) -> serverStopped
+    Just (Left (SomeException err)) -> serverCrashed $ show err
+  unsetEnv "COFFER_SERVER_PORT"
+
+unit_run_with_cmd_option_port :: IO ()
+unit_run_with_cmd_option_port = testServer
+  ""
+  ["--port=" ++ testPort]
+  (pure ())
+  (assertFailure "Server ended it's work")
+  (\err -> assertFailure ("Server raised exception : " ++ show err))
+
+-- can we hide stdout?
```

Comment:
Seems I'm running into issue with `optparse-applicative` even if I'm using silently. It's just prints help msg through silence

Reply:
Seems I'm running into issue with `optparse-applicative` even if I'm using silently. It's just prints help msg through silence 

I seems  I can do it with `execParserPure` and "manually" process result

Reply:
Maybe optparse-applicative is printing to stdout instead of stderr? The `silence` function only handles stdout. Try silencing both with `hSilence` instead.

Reply:
Ah! Yeah. Totally forgot about stderr :'D

Reply:
done